vmm/src/lib.rs

```diff
@@ -919,6 +928,56 @@ impl Vmm {
                 )));
             }
 
+            // Try at most 5 passes of dirty memory sending
+            const MAX_DIRTY_MIGRATIONS: usize = 5;
+            for i in 0..MAX_DIRTY_MIGRATIONS {
+                info!("Dirty memory migration {} of {}", i, MAX_DIRTY_MIGRATIONS);
+                // Send (dirty) memory table
+                let table = vm.dirty_memory_range_table()?;
+
+                // But if there are no regions go straight to pause
+                if table.regions().is_empty() {
+                    break;
+                }
+
+                Request::memory(table.length())
+                    .write_to(&mut socket)
+                    .unwrap();
+                table.write_to(&mut socket)?;
+                // And then the memory itself
+                vm.send_memory_regions(&table, &mut socket)?;
+                let res = Response::read_from(&mut socket)?;
+                if res.status() != Status::Ok {
+                    warn!("Error during dirty memory migration");
+                    Request::abandon().write_to(&mut socket)?;
+                    Response::read_from(&mut socket).ok();
+                    return Err(MigratableError::MigrateSend(anyhow!(
+                        "Error during dirty memory migration"
+                    )));
+                }
+            }
+
+            // Now pause VM
+            vm.pause()?;
+
+            // Send (dirty) memory table one last time
+            let table = vm.dirty_memory_range_table()?;
+            Request::memory(table.length())
+                .write_to(&mut socket)
+                .unwrap();
+            table.write_to(&mut socket)?;
+            // And then the memory itself
+            vm.send_memory_regions(&table, &mut socket)?;
+            let res = Response::read_from(&mut socket)?;
+            if res.status() != Status::Ok {
+                warn!("Error during dirty memory migration");
+                Request::abandon().write_to(&mut socket)?;
+                Response::read_from(&mut socket).ok();
+                return Err(MigratableError::MigrateSend(anyhow!(
+                    "Error during dirty memory migration"
+                )));
+            }
+
```

Comment:
This is a duplicate of the piece of code inside the above 5 iterations loop, maybe we could factorize that one.